src/main/java/org/skife/jdbi/v2/sqlobject/ParameterBinderRegistry.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.skife.jdbi.v2.sqlobject;
+
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+class ParameterBinderRegistry
+{
+    static final ParameterBinderFactory DEFAULT_BINDER_FACTORY = new ParameterBinderFactory() {
+        @Override
+        public boolean accepts(Class type, Method method, int param_idx) {
+            return true;
+        }
+
+        @Override
+        public Binder binderFor(Class type, Method method, int param_idx) {
+            return new PositionalBinder(param_idx);
+        }
```

Comment:
Yes, that default breaks when `-parameters` are not present on the javac command line.  It might be better to put that into a factory that does not accept anything that doesn't have parameters present, and then have the positional as the last in the chain.  Positional can be the default that is never removed, and the java 8 parameters a single item in the normal chain since it can turn on only when appropriate.